components/sup/src/manager/file_watcher.rs

```diff
@@ -2974,12 +2999,21 @@ mod tests {
             iterations
         }
 
-        fn spin_watcher(&self, setup: &mut WatcherSetup, iterations: u32) {
+        fn spin_watcher(&self, setup: &mut WatcherSetup, expected_event_count: u32) {
             let mut iteration = 0;
 
-            // After switching single_iteration() from recv() to try_recv(), this sleep is required
-            // for these tests to pass.
-            thread::sleep(Duration::from_secs(3));
+            let is_poll_watcher = match &setup.watcher.get_mut_underlying_watcher().real_watcher {
+                SupWatcher::Native(_watcher) => false,
+                SupWatcher::Fallback(_watcher) => true,
+            };
+
+            let mut iterations = expected_event_count;
+            if is_poll_watcher {
+                thread::sleep(Duration::from_secs(15));
```

Comment:
15 seconds seems really high. Did we play much with smaller delays?

Reply:
I originally got things working with a 30 second delay and then reduced it to 15 seconds and did not find it to be a problem. I can try. Will reduce if it will continue to pass the tests.